core/modules/comment/comment.install

```diff
@@ -66,6 +66,9 @@ function comment_uninstall() {
   if (db_field_exists('node', 'comment_close_override')) {
     db_drop_field('node', 'comment_close_override');
   }
+  // Flush outdated cache data.
+  // @todo Be more specific re cache prefix.
+  cache('cache')->flush();
```

Comment:
In https://github.com/backdrop/backdrop/pull/3514 (for https://github.com/backdrop/backdrop/issues/4904) we chose to clear caches via `backdrop_register_shutdown_function()` via an implementation of `hook_modules_uninstalled()` in the .module file instead of in the `hook_uninstall()` implementation in .install. Out of curiosity, why not do the same here?

Also, shouldn't we be making sure that this clearing of caches is done at the end of this function here? (after the deletion of the view config file)

Reply:
The difference is the context in which the flushing is done.

The entity module acts on changes done by other modules, so it's important that flushing runs last (in the shutdown function). Otherwise it still might pick up outdated info.
The comment module acts on changes done in it's own uninstall hook.